feature: Allow loading config values from secrets

maweil opened the pull request:
This PR adds the ability to load configuration values (and the superuser password) from files mounted in `/run/secrets/MUMBLE_CONFIG_*`files. These are usually created by docker or podman secrets.


I implemented this for my personal use in response to reading issue #27 but decided optimizing it for the default mountpoint of podman secrets, which I tested it with. 

Any feedback is appreciated, I plan to add some documentation to the README before marking this PR as *Ready for Review*.

maweil committed:
feature: Allow loading config values from secrets

- Check /run/secrets/MUMBLE_CONFIG_* and handle their content

Krzmbrzl commented on entrypoint.sh:
```diff
@@ -74,7 +74,7 @@ if [[ -f "$MUMBLE_CUSTOM_CONFIG_FILE" ]]; then
 else
 	# Ensures the config file is empty, starting from a clean slate
-	echo -e "# Config file automatically generated from the MUMBLE_CONFIG_* environment variables\n" > "${CONFIG_FILE}"
+	echo -e "# Config file automatically generated from the MUMBLE_CONFIG_* environment variables or secrets in /run/secrets/MUMBLE_CONFIG_* files\n" > "${CONFIG_FILE}"
```
Imo your addendum should simply go into its own line. That would make it a bit clearer, I think

maweil replied:
Thanks for the feedback. Fixed with c0ac92af445454e3850827a504baab9f82fa6cd3

Krzmbrzl commented on entrypoint.sh:
```diff
@@ -129,8 +146,12 @@ fi
+	#Variable to change the superuser password
+    "${server_invocation[@]}" -supw "$(cat /run/secrets/MUMBLE_SUPERUSER_PASSWORD)"
+    echo "Successfully configured superuser password from container secret"
+elif [[ -n "${MUMBLE_SUPERUSER_PASSWORD}" ]]; then
```
How about only setting the `MUMBLE_SUPERUSER_PASSWORD` variable here and keep the modification of the run command in the branch it already happens. Then we don't have to duplicate this functionality

maweil replied:
Thanks for the feedback, sounds good. Fixed in commit c0ac92af445454e3850827a504baab9f82fa6cd3 as well.

Krzmbrzl reviewed (commented):
Just some small things

maweil committed:
fix: Address code review comments

- Split comment in config file into two lines
- De-duplicate incovation for setting superuser password

maweil committed:
docs: Add example for running with podman secrets

- Update README.md

maweil commented:
Hi @Krzmbrzl,

thanks for the review. I have addressed your comments and added an example for running the container with secrets using `podman`. Is this sufficient or do we need an example for using it with docker as well?

Krzmbrzl reviewed (approved):
LGTM

Krzmbrzl commented:
> added an example for running the container with secrets using podman. Is this sufficient or do we need an example for using it with docker as well?

Given that you linked to the docs, I think it should be fine as it is. If not, we can always add an additional example later.
(I'm assuming that both options work kinda similar?)

Krzmbrzl commented:
Thanks for implementing this :+1:
